lib/core/src/persist/mod.rs

```diff
@@ -104,20 +106,21 @@ impl Persister {
     pub(crate) fn insert_or_update_payment_with_wallet_tx(&self, tx: &WalletTx) -> Result<()> {
         let tx_id = tx.txid.to_string();
         let is_tx_confirmed = tx.height.is_some();
-        let amount_sat = tx
-            .balance
-            .iter()
-            .filter_map(|(asset_id, balance)| {
-                if *asset_id == lwk_wollet::elements::AssetId::LIQUID_BTC {
-                    return Some(balance);
-                }
-                None
-            })
-            .sum::<i64>();
-        if amount_sat == 0 {
-            log::warn!("Attempted to persist a payment with no output amount: tx_id {tx_id}");
-            return Ok(());
+        let mut tx_balances = tx.balance.clone();
+        // Remove the Liquid Bitcoin asset balance
+        let mut balance = tx_balances
+            .remove(&AssetId::LIQUID_BTC)
+            .map(|balance| (LIQUID_BTC_ASSET_ID.to_string(), balance));
+        // If the balances are still not empty pop the asset balance
+        if tx_balances.is_empty().not() {
+            balance = tx_balances
+                .pop_first()
```

Comment:
Do we want to exclude liquid btc from the list of balances? Isn't it more valuable to have a complete list here?

Reply:
Here I meant only to ask if you think we should include the btc balance also as AssetBalance in the sequence returned. Currently it is not included.